### ISSUE TITLE
Shared link URL breaks when interacting with dashboard filters or 'details' menus

### ISSUE THREAD

**Issue by RobertJoonas** [plausible/analytics#2271]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Plausible Cloud from plausible.io

### Describe the bug

Here's an example from my own Plausible test site: 

1. Go to https://plausible.io/share/test2.net?auth=lyIYFDPmE15ylq2l_fsBS
2. Click on the "Filter" button, select any property and apply a random filter
3. Close the filter menu
4. Refresh the page 

  - After step 2, the URL is changed (removing the "/share" part). That means it becomes a regular dashboard link with just the `?auth=lyIYFDPmE15ylq2l_fsBS` query string.
  - Without account access to the dashboard, after step 4, you can still see the dashboard, but all stat reports will return a 404. You'll only see some spinning circles (see screenshot below).
- If you manually rewrite the "/share" part, it will start fetching the stats again.

So the issue only seems to be in ditching the "/share" string from the URL.

Actions other than interacting with the filter menu, that cause the same behavior:

- Opening any 'details' section (e.g. Top Pages > Details)
- Clicking on applied filter labels

### Expected behavior

The shared link URL should stay the same when doing the actions described.

### Screenshots

![image](https://user-images.githubusercontent.com/56999674/192219008-f37c7a3a-030a-4cc4-8cf8-de240125b81d.png)

### Environment

```markdown
any
```

**Comment by RobertJoonas** [comment]
> Is it a duplicate of https://github.com/plausible/analytics/issues/1038 ?

@ukutaht you're right. Didn't manage to find that one. I guess we can close this new one as there isn't much more context here that the other one doesn't contain.

I gave a more specific title to the initial issue so that it could be found easier next time.
